base/deprecated.jl

```diff
@@ -1009,4 +1009,18 @@ export @vectorize_1arg, @vectorize_2arg
 @deprecate abs(M::SymTridiagonal) abs.(M)
 @deprecate abs(x::AbstractSparseVector) abs.(x)
 
+# #18218
+eval(Base.LinAlg, quote
+    function arithtype(T)
+        depwarn("arithtype is deprecated as it is no longer needed in Julia Base.",
```

Comment:
With this, I mean that there is no need now for `arithtype` or anything like it, which used to operate on types. Now its functionality (but not the method itself) is replaced by the definition and use of `matprod`.